File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>
#include <stack>
using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    stack<int> deploy;
    for (int i = progresses.size() - 1; i >= 0; i--) {
        int day = (100 - progresses[i]) / speeds[i];
        if ((100 - progresses[i]) % speeds[i] != 0) day++;
        deploy.push(day);
    }
    int tmp = 1;
    int top;

    for (int i = 0; i < progresses.size() - 1; i++) {
        top = deploy.top();
        deploy.pop();

        if (deploy.top() > top) {
            answer.push_back(tmp);
            tmp = 1;
        }
        else {
            deploy.pop();
            deploy.push(top);
            tmp++;
        }
    }
    answer.push_back(tmp);
    
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        ArrayList<Integer> answer = new ArrayList<>();
        
        int deployDay = 0;
        int count = 0;
        
        for (int i = 0; i < progresses.length; i++) {
            int remain = 100 - progresses[i];
            int day = (remain + speeds[i] - 1) / speeds[i];
            
            if (day > deployDay) {
                if (count > 0) {
                    answer.add(count);
                }
                
                deployDay = day;
                count = 1;
            } else {
                count++;
            }
        }
        answer.add(count);
        
        return answer.stream()
                .mapToInt(Integer::intValue)
                .toArray();
    }
}